src/main/java/uk/nhs/tis/trainee/notifications/service/NotificationService.java

```diff
@@ -288,6 +289,44 @@ private UserDetails getTraineeDetails(String personId) {
     }
   }
 
+  /**
+   * Check whether an object meets the selected notification criteria.
+   *
+   * @param programmeMembership The programme membership to check.
+   * @param checkNewStarter     Whether the trainee must be a new starter.
+   * @param checkPilot          Whether the trainee must be in a pilot.
+   * @return true if all criteria bet, or false if one or more criteria fail.
+   */
+  public boolean meetsCriteria(ProgrammeMembership programmeMembership,
+      boolean checkNewStarter, boolean checkPilot) {
+    String traineeId = programmeMembership.getPersonId();
+    String pmId = programmeMembership.getTisId();
+
+    if (checkNewStarter) {
+      String apiPath = "/api/programme-membership/isnewstarter/{traineeId}/{pmId}";
+      Boolean isNewStarter = restTemplate.getForObject(serviceUrl + apiPath, Boolean.class,
```

Comment:
If the API call fails (e.g. if trainee-details is down), I presume not wrapping this in a try-catch will mean the exception that is thrown will requeue the original message. In my branch I was returning a default 'false', but I think I prefer this behaviour.

Reply:
🤔 I'll have to double check that

Reply:
I think I misunderstood your comment originally. I thought you were seeing it default to false and discarding the message when the details service was down 😬

Reply:
Ah no, sorry - bad phrasing. You're right: if the details service is down then it should requeue the message. The work I was doing in parallel had been trapping any REST errors and discarding the message, but I'm going to amend it to match your code.